src/nfa/operations.cc

```diff
@@ -927,27 +927,54 @@ Nfa mata::nfa::algorithms::minimize_brzozowski(const Nfa& aut) {
     return determinize(revert(determinize(revert(aut))));
 }
 
-Nfa mata::nfa::minimize(
-                const Nfa& aut,
-                const ParameterMap& params)
-{
-    Nfa result;
-    // setting the default algorithm
-    decltype(algorithms::minimize_brzozowski)* algo = algorithms::minimize_brzozowski;
+Nfa mata::nfa::make_minimal_dfa(const Nfa& nfa, const ParameterMap& params) {
     if (!haskey(params, "algorithm")) {
         throw std::runtime_error(std::to_string(__func__) +
             " requires setting the \"algorithm\" key in the \"params\" argument; "
             "received: " + std::to_string(params));
     }
 
+    // Setting the algorithm. Default is Hopcroft.
     const std::string& str_algo = params.at("algorithm");
-    if ("brzozowski" == str_algo) {  /* default */ }
+    decltype(algorithms::minimize_brzozowski)* algo = algorithms::minimize_brzozowski;
+    if (str_algo == "brzozowski") { /* default */ }
     else {
+        throw std::runtime_error(std::string(__func__) +
+            " received an unknown value for the \"algorithm\" key: " + str_algo);
+    }
+
+    return algo(nfa);
+}
+
+Nfa mata::nfa::minimize(const Nfa &dfa, const ParameterMap& params)
+{
+    if (!haskey(params, "algorithm")) {
         throw std::runtime_error(std::to_string(__func__) +
-            " received an unknown value of the \"algorithm\" key: " + str_algo);
+            " requires setting the \"algorithm\" key in the \"params\" argument; "
+            "received: " + std::to_string(params));
     }
 
-    return algo(aut);
+    // Setting the algorithm. Default is Hopcroft.
+    const std::string& str_algo = params.at("algorithm");
+    decltype(algorithms::minimize_hopcroft)* algo = algorithms::minimize_hopcroft;
+    if (str_algo == "hopcroft") {
+        /* default */;
+    } else if (str_algo == "brzozowski") {
+        algo = algorithms::minimize_brzozowski;
```

Comment:
Maybe this should be named `make_minimal_dfa_brzozowski()`? To show it can be used for NFAs.

Reply:
But this will break the naming convention for minimizazion algorithms.

Reply:
True, but Brzozowski does not perform minimization as per the definition given by Lukáš. But this is debatable. What does @jurajsic think?

Reply:
I do not know, but I realized that this change, that minimize now HAVE to take DFA, might break noodler (and anyone using mata, if such people exists), because before it was assumed it takes nfa and calls brzozowski.